staticd/static_vrf.c

```diff
@@ -150,24 +150,16 @@ struct static_vrf *static_vrf_lookup_by_name(const char *name)
 
 static int static_vrf_config_write(struct vty *vty)
 {
-	struct vrf *vrf;
-
-	RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name) {
-		if (vrf->vrf_id != VRF_DEFAULT)
-			vty_frame(vty, "vrf %s\n", vrf->name);
+	struct lyd_node *dnode;
+	int written = 0;
 
-		static_config(vty, vrf->info, AFI_IP,
-			      SAFI_UNICAST, "ip route");
-		static_config(vty, vrf->info, AFI_IP,
-			      SAFI_MULTICAST, "ip mroute");
-		static_config(vty, vrf->info, AFI_IP6,
-			      SAFI_UNICAST, "ipv6 route");
-
-		if (vrf->vrf_id != VRF_DEFAULT)
-			vty_endframe(vty, "exit-vrf\n!\n");
+	dnode = yang_dnode_get(running_config->dnode, "/frr-routing:routing");
```

Comment:
The node chosen here seems very shallow in the tree, this probably works b/c its traversing a `libyang` tree inside `staticd`. Wondering, if this code has to move with newer centralized management will it break? Is there a more specific node that could be chosen?

Reply:
I don't see any problem that can happen when this is moved to mgmtd. Yes, using this path means that it'll output the whole config, not only the staticd config, but this is the point of mgmtd :)